src/image_gen/imagegen_init.cpp

```diff
@@ -56,8 +135,82 @@ std::variant<Status, ImageGenPipelineArgs> prepareImageGenPipelineArgs(const goo
+            return std::get<Status>(res);
+        }
+        args.staticReshapeSettings = StaticReshapeSettingsArgs(std::get<std::vector<resolution_t>>(res));
+
+        if (isNPU && args.staticReshapeSettings.value().resolution.size() > 1) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "NPU cannot have multiple resolutions in static settings");
+            return StatusCode::SHAPE_DYNAMIC_BUT_NPU_USED;
+        }
+    } else if (isNPU) {
+        SPDLOG_LOGGER_ERROR(modelmanager_logger, "Cannot use NPU without setting static resolution");
+        return StatusCode::SHAPE_DYNAMIC_BUT_NPU_USED;
+    }
+    if (args.staticReshapeSettings.has_value()) {
+        if (nodeOptions.has_max_resolution()) {  // non default
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Cannot explicitly use max resolution when using static settings");
+            return StatusCode::STATIC_RESOLUTION_MISUSE;
+        }
+        if (nodeOptions.has_num_images_per_prompt()) {
+            if (args.staticReshapeSettings.value().resolution.size() > 1) {
+                SPDLOG_LOGGER_ERROR(modelmanager_logger, "Cannot use static num images per prompt with multiple resolutions");
+                return StatusCode::STATIC_RESOLUTION_MISUSE;
+            }
+            args.staticReshapeSettings->numImagesPerPrompt = nodeOptions.num_images_per_prompt();
+        }
+        if (nodeOptions.has_guidance_scale()) {
+            if (args.staticReshapeSettings.value().resolution.size() > 1) {
+                SPDLOG_LOGGER_ERROR(modelmanager_logger, "Cannot use static guidance scale with multiple resolutions");
+                return StatusCode::STATIC_RESOLUTION_MISUSE;
+            }
+            args.staticReshapeSettings->guidanceScale = nodeOptions.guidance_scale();
+        }
+        if (args.staticReshapeSettings.value().resolution.size() == 1 && nodeOptions.has_max_num_images_per_prompt()) {  // non default
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Cannot explicitly use max num images per prompt when using static settings");
+            return StatusCode::STATIC_RESOLUTION_MISUSE;
+        }
+        if (args.staticReshapeSettings.value().resolution.size() == 1 && nodeOptions.has_max_resolution()) {  // non default
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Cannot explicitly use max resolution when using static settings");
+            return StatusCode::STATIC_RESOLUTION_MISUSE;
+        }
+    } else {
+        if (nodeOptions.has_guidance_scale()) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Cannot explicitly use static guidance scale when not using static resolution");
+            return StatusCode::STATIC_RESOLUTION_MISUSE;
+        }
+        if (nodeOptions.has_num_images_per_prompt()) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Cannot explicitly use static num images per prompt when not using static resolution");
+            return StatusCode::STATIC_RESOLUTION_MISUSE;
+        }
```

Comment:
This whole function became quite big with multiple conditionals. Maybe extracting some validation parts would help readability?

Reply:
I insist on keeping it as it is. Most of the options affect each other, I think spliting it wouldn't add much value.